fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingScrubMissing.java

```diff
@@ -217,16 +217,10 @@ public IndexingScrubMissing(@Nonnull final IndexingCommon common,
                     final byte[] keyBytes = maintainer.getIndexSubspace().pack(valueKey);
                     return maintainer.state.transaction.get(keyBytes).thenApply(indexVal -> indexVal == null ? valueKey : null);
                 })
+                .filter(Objects::nonNull)
                 .collect(Collectors.toList()))
-                .thenApply(list -> {
-                    Tuple missingKey = null;
-                    for (Tuple item: list) {
-                        if (item != null) {
-                            missingKey = item;
-                        }
-                    }
-                    if (missingKey  == null) {
-                        // no null index(s) = no record to index
+                .thenApply(missingIndexesKeys -> {
+                    if (missingIndexesKeys.isEmpty() || (missingIndexesKeys.size() == 1 && missingIndexesKeys.get(0) == null)) {
```

Comment:
Since the `filter()` wasn't really filtering, this will be a list of `null`'s for entries that are present and non-`null`'s for entries that were missing, so your test needs to be whether or not there is a non-null entry in the list.